internal/events/types/hype_train/hype_train_event.go

```diff
@@ -0,0 +1,224 @@
+
+
+	if params.Trigger == "hype-train-begin" {
+		switch params.Transport {
+			case models.TransportEventSub:
+				body := *&models.EventsubResponse{
+					Subscription: models.EventsubSubscription{
+						ID:      params.ID,
+						Status:  "enabled",
+						Type:    triggerMapping[params.Transport][params.Trigger],
+						Version: "1.0",
+						Condition: models.EventsubCondition{
+							BroadcasterUserID: params.ToUserID,
+						},
+						Transport: models.EventsubTransport{
+							Method:   "webhook",
+							Callback: "null",
+						},
+						CreatedAt: util.GetTimestamp().Format(time.RFC3339Nano),
+					},
+					Event: models.HypeTrainEventSubEvent{
+						BroadcasterUserID:    	params.ToUserID,
+						BroadcasterUserLogin: 	params.ToUserName,
+						BroadcasterUserName:  	params.ToUserName,
+						Total:               	localTotal,
+						Progress:            	localProgress,
+						Goal:             		localGoal,
+						TopContributions:    	localTC,
+						LastContribution: 		localLC,
+						StartedAtTimestamp:  	util.GetTimestamp().Format(time.RFC3339Nano),
+						ExpiresAtTimestamp:  	util.GetTimestamp().Format(time.RFC3339Nano),
+				},
+			}
+
+			event, err = json.Marshal(body)
+			if err != nil {
+				return events.MockEventResponse{}, err
+			}
+
+			default:
+			return events.MockEventResponse{}, nil
+		}
+	}
+
+	if params.Trigger == "hype-train-end" {
+		switch params.Transport {
+		case models.TransportEventSub:
+			body := *&models.EventsubResponse{
+				Subscription: models.EventsubSubscription{
+					ID:      params.ID,
+					Status:  "enabled",
+					Type:    triggerMapping[params.Transport][params.Trigger],
+					Version: "1.0",
+					Condition: models.EventsubCondition{
+						BroadcasterUserID: params.ToUserID,
+					},
+					Transport: models.EventsubTransport{
+						Method:   "webhook",
+						Callback: "null",
+					},
+					CreatedAt: util.GetTimestamp().Format(time.RFC3339Nano),
+				},
+				Event: models.HypeTrainEventSubEvent{
+					BroadcasterUserID:    	 params.ToUserID,
+					BroadcasterUserLogin: 	 params.ToUserName,
+					BroadcasterUserName:  	 params.ToUserName,
+					Level:               	 util.RandomViewerCount()%4,
+					Total:             		 localTotal,
+					TopContributions:    	 localTC,
+					StartedAtTimestamp:  	 util.GetTimestamp().Format(time.RFC3339Nano),
+					EndedAtTimestamp: 	 	 util.GetTimestamp().Format(time.RFC3339Nano),
+					CooldownEndsAtTimestamp: util.GetTimestamp().Format(time.RFC3339Nano),
+				},
+			}
+
+			event, err = json.Marshal(body)
+			if err != nil {
+				return events.MockEventResponse{}, err
+			}
+		default:
+			return events.MockEventResponse{}, nil
+		}
+	}
+
+	if params.Trigger == "hype-train-progress" {
```

Comment:
This could certainly work here- I'll message you some thoughts more concretely around this.